client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts

```diff
@@ -0,0 +1,106 @@
+import { useSelector } from 'react-redux';
+import StoreStateType from 'redux/storeStateType';
+
+import axios from 'Utils/axios';
+import InteractedContact from 'models/InteractedContact';
+import IdentificationTypes from 'models/enums/IdentificationTypes';
+import InteractedContactFields from 'models/enums/InteractedContact';
+
+import { useContactQuestioningOutcome, useContactQuestioningParameters } from './ContactQuestioningInterfaces';
+
+const useContactQuestioning = (parameters: useContactQuestioningParameters): useContactQuestioningOutcome => {
+    const { interactedContactsState, setCurrentInteractedContact } = parameters;
+
+    const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
+
+    const saveContact = (interactedContact: InteractedContact) => {
+        updateInteractedContact(interactedContact, InteractedContactFields.EXPAND, false);
+        const contacts = [interactedContact];
+        axios.post('/contactedPeople/interactedContacts',
+        {
+            unSavedContacts: { contacts }
+        });
+    };
+
+    const saveContactQuestioning = (): Promise<void> => {
+        const contacts = interactedContactsState.interactedContacts;
+
+        return axios.post('/contactedPeople/interactedContacts',
+            {
+                unSavedContacts: { contacts }
+            }
+        );
+    };
+
+    const loadInteractedContacts = () => {
+        let interactedContacts: InteractedContact[] = [];
+
+        axios.get('/contactedPeople/' + epidemiologyNumber).then((result: any) => {
```

Comment:
why do the same query twice? 🤔 
if the first one is just to get a boolean indication if there are contacts - 
maybe query count of contact people to make the response lighter and faster
or pass the response from the first one down via props or context?

Reply:
because i need to check if there are contacts to show the relevant tabs.

Reply:
I want all the contacts loaded only if the tab was chosen.

Reply:
yes but to check if you should show the tab or not - youre doing the same query, right?

Reply:
right

Reply:
If so, you get the same result twice... 
you could just get the count() of the contectedPeople and then if count > 0 display the tab
or do the same as now, except instead of re-fetching inside useContactQuestioning again, just pass the info down with state or props

Reply:
I will do a new query
query {
  allContactedPeople(filter: {contactEventByContactEvent: {investigationId: {equalTo: 122}}}) {
    totalCount
  }
}